File: html/redcap_v15.7.6/API/examples/csharp/DeleteFile.cs
using RestSharp;
namespace RedcapCSharpApiExamples
{
    public static class DeleteFile
    {
        public static async Task<RestResponse> DeleteFileAsync(string field, string theEvent, string record)
        {
            try
            {
                // Setup
                var apiToken = Config.ApiToken;
                var uri = new Uri(Config.ApiUrl);

                // Create Request
                var client = new RestClient();
                var request = new RestRequest(uri, Method.Post);
                request.AddHeader("Content-Type", "application/x-www-form-urlencoded");
                request.AddParameter("token", apiToken);
                request.AddParameter("content", "file");
                request.AddParameter("action", "delete");
                request.AddParameter("field", field);
                request.AddParameter("event", theEvent);
                request.AddParameter("record", record);
                request.AddParameter("returnFormat", "json");

                // Execute Request
                var response = await client.ExecuteAsync(request);
                return response;
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
                throw;
            }
        }
    }
}

File: html/redcap_v15.7.6/API/examples/ruby/delete_file.rb
#!/usr/bin/env ruby

require 'curl'
require './settings.rb'
include Settings

fields = {
  :token => Settings::API_TOKEN,
  :content => 'file',
  :action => 'delete',
  :record => 'f21a3ffd37fc0b3c',
  :field => 'file_upload',
  :event => 'event_1_arm_1',
}

fields[:returnFormat] = 'json'

ch = Curl::Easy.http_post(
  Settings::API_URL,
  fields.collect{|k, v| Curl::PostField.content(k.to_s, v)}
)
puts ch.body_str
